### PR TITLE
KITE-1095: Continue iteration over additional file system partitions if a path without files is found.

### PR DESCRIPTION
**Pull request by scheeser**

### CODE REVIEW

File: kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/PathIterator.java [hunk 1]
```diff
@@ -112,11 +112,13 @@ private boolean advance() {
           nextFileSet.add(stat.getPath());
         }
       }
+
       if (nextFileSet.size() > 0) {
         this.files = nextFileSet.iterator();
         return true;
+      } else if (partitions == null) {
+          return false;
```
Comment (rdblue) [inline kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/PathIterator.java]: Nit: indentation is off.

**Comment by rdblue** [comment]
+1

Thanks for contributing, @scheeser!

**Comment by scheeser** [comment]
@rdblue I've updated the pull request with the corrected indentation.

**Comment by mkwhitacre** [comment]
+1.  Re-kicked off travis build b/c it seemed to fail b/c it got stuck downloading dependencies vs valid reason.

**Comment by rbrush** [comment]
+1 as well. I will merge this if there are no objections; I see @rdblue and @mkwhitacre have +1's above.

**Comment by mkwhitacre** [comment]
Didn't get any objections so merging.

**Pull request merged**